lib/src/test.nr

```diff
@@ -0,0 +1,53 @@
+    ];
+    let depth = 0x02;
+    let value = [
+        0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xba, 0x52, 0xc7, 0x57, 0x64, 0xd6, 0xf5, 0x94, 0x73, 0x5d, 0xc7, 0x35, 0xbe, 0x7f, 0x18, 0x30, 0xcd, 0xf5, 0x8d, 0xdf
+    ];
+    let invalid_storage_root = [0; 32];
+
+    let trie_proof = TrieProof { key, proof, depth, value };
+    let _ = trie_proof.verify_storage_root(invalid_storage_root);
+}
+
+#[test(should_fail_with="Terminal node hash does not match the hash extracted from the preceding node")]
+fn test_assertion_message_for_invalid_leaf() {
+    let key = [0; 32];
+    let proof = [
```

Comment:
Right, fixed. I've added also CI configuration to this PR so that tests are automatically run every PR. Here how it looks in our fork https://github.com/vlayer-xyz/noir-trie-proofs/actions/runs/8230840524/job/22504938958